### PR TITLE
services: fix binary logging empty metadata NPE

### PR DESCRIPTION
**Pull request by zpencer**
Also added test to cover this case, and renamed test variables
to make it more obvious in the future that the default metadata
is non empty.

### CODE REVIEW

File: services/src/test/java/io/grpc/services/BinaryLogTest.java [hunk 1]
```diff
@@ -371,6 +371,15 @@ public void socketToProto_unknown() throws Exception {
         BinaryLog.socketToProto(unknownSocket));
   }
 
+  @Test
+  public void metadataToProto_empty() throws Exception {
+    assertEquals(
+        io.grpc.binarylog.Metadata
```
Comment (carl-mastrangelo) [inline services/src/test/java/io/grpc/services/BinaryLogTest.java]: `io.grpc.binarylog.Metadata.getDefaultInstance()`

File: services/src/main/java/io/grpc/services/BinaryLog.java [hunk 2]
```diff
@@ -416,10 +416,10 @@ static Peer socketToProto(SocketAddress address) {
   static io.grpc.binarylog.Metadata metadataToProto(Metadata metadata, int maxHeaderBytes) {
     Preconditions.checkState(maxHeaderBytes >= 0);
     Builder builder = io.grpc.binarylog.Metadata.newBuilder();
-    if (maxHeaderBytes > 0) {
-      byte[][] serialized = InternalMetadata.serialize(metadata);
+    // This code is tightly coupled with Metadata's implementation
+    byte[][] serialized;
+    if (maxHeaderBytes > 0 && (serialized = InternalMetadata.serialize(metadata)) != null) {
```
Comment (sonyakc) [inline services/src/main/java/io/grpc/services/BinaryLog.java]: suggestion: I'd add a null check for ```metadata``` before setting ```serialized ```

Comment (zpencer) [reply services/src/main/java/io/grpc/services/BinaryLog.java]: Good idea

File: services/src/test/java/io/grpc/services/BinaryLogTest.java (hunk 1 above)
Comment (zpencer) [reply services/src/test/java/io/grpc/services/BinaryLogTest.java]: done

**Pull request merged**